### PR TITLE
Implement some INI parsing bits

### PR DESCRIPTION
**Pull request by jonwil**

### CODE REVIEW

File: src/game/common/ini/ini.cpp [hunk 1]
```diff
@@ -74,23 +76,21 @@ const BlockParse TheTypeTable[] = {
     {"AIData", HOOK_BLOCK(0x00518F00) /*&INI::parseAIDataDefinition*/},
     {"Animation", &Anim2DCollection::Parse_Anim2D_Definition /*&INI::parseAnim2DDefinition*/},
     {"Armor", HOOK_BLOCK(0x004B60A0) /*&INI::parseArmorDefinition*/},
-    //{"AudioEvent", HOOK_BLOCK(0x0044ED70) /*&INI::parseAudioEventDefinition*/},
     {"AudioEvent", AudioEventInfo::Parse_Audio_Event_Definition},
     {"AudioSettings", &AudioSettings::Parse_Audio_Settings_Definition},
     {"Bridge", &TerrainRoadCollection::Parse_Terrain_Bridge_Definition},
     {"Campaign", HOOK_BLOCK(0x00517490) /*&INI::parseCampaignDefinition*/},
-    {"ChallengeGenerals", HOOK_BLOCK(0x005170B0) /*&INI::parseChallengeModeDefinition*/},
+    {"ChallengeGenerals", ChallengeGenerals::Parse_Challenge_Mode_Definition},
```
Comment (feliwir) [inline src/game/common/ini/ini.cpp:83]: Add & before the function

File: src/game/common/ini/ini.cpp [hunk 2]
```diff
@@ -74,23 +76,21 @@ const BlockParse TheTypeTable[] = {
...
     {"CommandButton", HOOK_BLOCK(0x00516CE0) /*&INI::parseCommandButtonDefinition*/},
     {"CommandMap", HOOK_BLOCK(0x00498480) /*&INI::parseMetaMapDefinition*/},
     {"CommandSet", HOOK_BLOCK(0x00516CD0) /*&INI::parseCommandSetDefinition*/},
     {"ControlBarScheme", HOOK_BLOCK(0x00516BA0) /*&INI::parseControlBarSchemeDefinition*/},
-    {"ControlBarResizer", HOOK_BLOCK(0x0062D610) /*&INI::parseControlBarResizerDefinition*/},
+    {"ControlBarResizer", INI::Parse_Control_Bar_Resizer_Definition},
```
Comment (feliwir) [inline src/game/common/ini/ini.cpp]: Add & before the function

File: src/game/network/gamespy/chat.cpp [hunk 3]
```diff
@@ -0,0 +1,59 @@
+/**
+ * @file
+ *
+ * @author Jonathan Wilson
+ *
+ * @brief Gamespy stuff
```
Comment (xezon) [inline src/game/network/gamespy/chat.cpp:6]: GameSpy Chat

File: src/game/common/audio/audioeventinfo.cpp [hunk 4]
```diff
@@ -103,6 +103,45 @@ void AudioEventInfo::Parse_Audio_Event_Definition(INI *ini)
...
 }
 
+void AudioEventInfo::Parse_Music_Track_Definition(INI *ini)
+{
+    Utf8String track;
+    track.Set(ini->Get_Next_Token());
+    AudioEventInfo *info = g_theAudio->New_Audio_Event_Info(track);
+
+    if (info != nullptr) {
+        AudioEventInfo *default = g_theAudio->Find_Audio_Event_Info("DefaultMusicTrack");
```
Comment (xezon) [inline src/game/common/audio/audioeventinfo.cpp]: default_audio_event

File: src/game/common/audio/audioeventinfo.cpp [hunk 5]
```diff
@@ -103,6 +103,45 @@ void AudioEventInfo::Parse_Audio_Event_Definition(INI *ini)
...
+}
+
+void AudioEventInfo::Parse_Dialog_Definition(INI *ini)
+{
+    Utf8String track;
+    track.Set(ini->Get_Next_Token());
+    AudioEventInfo *info = g_theAudio->New_Audio_Event_Info(track);
+
+    if (info != nullptr) {
+        AudioEventInfo *default = g_theAudio->Find_Audio_Event_Info("DefaultDialog");
```
Comment (xezon) [inline src/game/common/audio/audioeventinfo.cpp]: default_audio_event

File: src/game/common/audio/audioeventinfo.cpp [hunk 6]
```diff
@@ -103,6 +103,45 @@ void AudioEventInfo::Parse_Audio_Event_Definition(INI *ini)
...
+        info->m_eventType = EVENT_MUSIC;
+        ini->Init_From_INI(info, s_audioEventParseTable);
+    }
+}
+
+void AudioEventInfo::Parse_Dialog_Definition(INI *ini)
+{
+    Utf8String track;
+    track.Set(ini->Get_Next_Token());
+    AudioEventInfo *info = g_theAudio->New_Audio_Event_Info(track);
```
Comment (xezon) [inline src/game/common/audio/audioeventinfo.cpp]: new_audio_event

File: src/game/common/audio/audioeventinfo.cpp [hunk 7]
```diff
@@ -103,6 +103,45 @@ void AudioEventInfo::Parse_Audio_Event_Definition(INI *ini)
     }
 }
 
+void AudioEventInfo::Parse_Music_Track_Definition(INI *ini)
+{
+    Utf8String track;
+    track.Set(ini->Get_Next_Token());
+    AudioEventInfo *info = g_theAudio->New_Audio_Event_Info(track);
```
Comment (xezon) [inline src/game/common/audio/audioeventinfo.cpp]: new_audio_event

File: src/game/common/ini/ini.cpp (hunk 1 above)
Comment (xezon) [reply src/game/common/ini/ini.cpp:83]: Makes no difference for function pointers.

Comment (feliwir) [reply src/game/common/ini/ini.cpp:83]: Yes, but we should keep it somewhat consistent here imo

Comment (jonwil) [reply src/game/common/ini/ini.cpp:83]: fixed.

File: src/game/common/ini/ini.cpp (hunk 2 above)
Comment (jonwil) [reply src/game/common/ini/ini.cpp]: fixed.

File: src/game/common/audio/audioeventinfo.cpp (hunk 4 above)
Comment (jonwil) [reply src/game/common/audio/audioeventinfo.cpp]: fixed.

File: src/game/common/audio/audioeventinfo.cpp (hunk 5 above)
Comment (jonwil) [reply src/game/common/audio/audioeventinfo.cpp]: fixed.

File: src/game/common/audio/audioeventinfo.cpp (hunk 6 above)
Comment (jonwil) [reply src/game/common/audio/audioeventinfo.cpp]: fixed.

File: src/game/common/audio/audioeventinfo.cpp (hunk 7 above)
Comment (jonwil) [reply src/game/common/audio/audioeventinfo.cpp]: fixed.

**Review by xezon — APPROVED** [review]

**Pull request merged**